src/WebJobs.Script.Extensibility/ScriptBindingContext.cs

```diff
@@ -107,12 +114,32 @@ public ScriptBindingContext(JObject bindingMetadata)
         public TValue GetMetadataValue<TValue>(string name, TValue defaultValue = default(TValue))
         {
             JToken value = null;
-            if (Metadata.TryGetValue(name, StringComparison.OrdinalIgnoreCase, out value))
+            try
             {
-                return value.Value<TValue>();
+                if (Metadata.TryGetValue(name, StringComparison.OrdinalIgnoreCase, out value))
+                {
+                    return value.Value<TValue>();
+                }
+            }
+            catch (FormatException e)
+            {
+                throw new FormatException($"Invalid value specified for binding property '{name}' of type {PrettyTypeName(typeof(TValue))}.", e);
             }
 
             return defaultValue;
         }
+
+        private static string PrettyTypeName(Type t)
```

Comment:
I see it as a best effort to guide the user to the right type as we have that information.  I think we'd ideally keep it as close to human readable as possible, hence prettification and `Name` instead of `FullName` (I don't think that the extra info it provides is worth the clutter/confusion)